### ISSUE TITLE
giving feedback

### ISSUE THREAD

**Issue by codefromthecrypt** [jcchavezs/wasm-tracing#2]
* when we have an abi that can only add a tag, this sounds more like configuration than tracing.
  * often programs are compiled from main, and certain weird setup to wasm can be contained there
* Even if it was a "current span" abi, it would be weird to brand it as http-wasm
  * tracing value is pliability and also it was discussed out of scope for http-wasm, so 2 things troubling.
* If this was about tracing vs key/value logging, why pin it to http-wasm, and why not define a reporter ABI instead?
  * then you can have a path from compiling something like a slimmed zipkin-go to it
  * wasi already has functions for RNG and nanotime, it just needs a reporter/handler/etc

while a neat experiment, I really hope this doesn't get anchored into being pushed to http-wasm again, just because envoy's sdk has something similar and this is similar to that.

**Comment by codefromthecrypt** [comment]
here are some ideas. however, I really think folks who are in otel etc should likely be hosting code like this eventually not http-wasm as they are entirely separate concerns

If it is too tall an order to think about reporting a span, you could make a "current span" api in a tracing package to toy with, which basically allows a key/value tag but more explicitly going into a span not a log.

the guest library side could have a normal "current span" that with setup code has a stripped down way that can only contribute to an existing one. someone can import that library into http-wasm or anything else.

later, think more through tracing with folks staffed on it, as surely someone can do this. we'd expect things like how AWS lamda works today to eventually reach wasm (e.g. special ENV vars to propagate a parent etc.) and just like lambda that sort of stuff can be setup code.

envoy's sdk is differnet than a general purpose ABI as it has lifecycle and normal SDK assumptions that don't carry through when a library not a product SDK. similar in k8s the scheduler abi is porting a whole SDK so it has different needs and can make different assumptions than a general purpose http handler.

If you disagree, which we might, one way to test the strength of this idea is to pitch it to otel or wasi-http maintainers, and if they would want a separate tracing thing for each wasi-X or not. Maybe I'm way off.

TL;DR; I personally think it is a good idea to have something like this, but a bad idea to anchor it as an http-wasm thing. All the other tracing things out there build on a foundation of some tracing substrate, because they work with other libraries who will have similar concerns.

**Comment by jcchavezs** [comment]
Thanks for the feedback. This will definitively going to change into something like wazero-tracing or wazero-tracing-enrichment. Originally my use case was about http-wasm but talking to @anuraaga I realized this does not belong to http-wasm realm as the only connection to http-wasm now is the e2e so I will change the name, the module and keep http-wasm as an example or e2e

**Comment by codefromthecrypt** [comment]
👍 I'm not sure it would make sense to brand this as wazero either. sounds like you are just making an ABI and it would want to work on other runtimes right? Maybe focus on what it does and not the brand umbrella is a good way to start, as not all projects would want to own generic ABIs
